src/app/beer_garden/router.py

```diff
@@ -84,29 +93,31 @@ def route(operation: Operation):
     Returns:
 
     """
+    logger.debug(f"Routing {operation}")
+
     operation = _pre_route(operation)
 
-    if should_forward(operation):
-        return initiate_forward(operation)
+    if not operation.operation_type:
+        raise RoutingRequestException(
+            f"Unknown operation type '{operation.operation_type}'"
+        )
 
-    return execute_local(operation)
+    # Then use that to determine which garden the operation is targeting
+    operation.target_garden_name = _determine_target_garden(operation)
 
+    # If it's targeted at THIS garden, execute
+    if operation.target_garden_name == config.get("garden.name"):
+        return execute_local(operation)
 
-def should_forward(operation: Operation) -> bool:
-    """Routing logic to determine if an operation should be forwardedl
+    # Otherwise, validate that it can be forwarded
+    if operation.operation_type not in routable_operations:
+        raise RoutingRequestException(
+            f"Operation type '{operation.operation_type}' can not be forwarded"
+        )
```

Comment:
Should you run this check before executing local?

Reply:
This is kind of the heart of this change. I want to inspect the operation and use it to determine which garden it SHOULD be routed to.

Then, if it turns out that the operation is not allowed to be routed, I can raise the correct exception here.

The alternative can be confusing. For example, say we decided to not allow forwarding for system delete events. If somebody tries to delete a system from a child garden and we did things the old way:
- We'd check if the operation was routable. It's not, so we'd attempt to execute it locally.
- That system doesn't exist locally, so we raise a "not found" exception
- The user is confused because the system is right there

This way allows us to be a little more accurate with the failure cases.